gherkin/php/src/GherkinParser.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Cucumber\Gherkin;
+
+use Cucumber\Gherkin\ParserException\CompositeParserException;
+use Cucumber\Messages\Envelope;
+use Cucumber\Messages\GherkinDocument;
+use Cucumber\Messages\Id\IdGenerator;
+use Cucumber\Messages\Id\IncrementingIdGenerator;
+use Cucumber\Messages\Id\UuidIdGenerator;
+use Cucumber\Messages\Location as MessageLocation;
+use Cucumber\Messages\ParseError;
+use Cucumber\Messages\Source;
+use Cucumber\Messages\Source\MediaType;
+use Cucumber\Messages\SourceReference;
+use Generator;
+
+final class GherkinParser
+{
+    private readonly PickleCompiler $pickleCompiler;
+    private readonly IdGenerator $idGenerator;
+
+    /**
+     * @param bool $predictableIds Ignored if IdGenerator is provided
+     * @param bool $includeSource
```

Comment:
Redundant params here can be removed (but add some description of the method as it's public-facing)